### ISSUE TITLE
Type issues from creating Legal Entity Association for soleProprietorship [BUG]

### ISSUE THREAD

**Issue by MD-Fleksa** [Adyen/adyen-node-api-library#1127]
**Describe the bug**
There's some types needs to be adjust inside the SDK. 

 1. When we don't mention the type to link the individual to soleProprietorship it's showing the type error. From the SDK side it should be optional.
 
![image](https://user-images.githubusercontent.com/93864853/226986158-a427c103-6abe-43d7-9b6e-e166dcba4f66.png)

 2. For linking the individual to **soleProprietorship** under the **entityAssociations** object there's no type **soleProprietorship** for us to mention. Because of this issue we're putting any for making it work.

![image (1)](https://user-images.githubusercontent.com/93864853/226986307-b1a4afe0-b05a-4be3-a33a-4462da3e7938.png)

**To Reproduce**
1.  Create a soleProprietorship entity.
![image](https://user-images.githubusercontent.com/93864853/226986726-d97a4669-20a0-4edc-8ff8-926d9283ff7c.png)

2. Create an individual entity.
![image](https://user-images.githubusercontent.com/93864853/226986900-0a65fef5-e773-4d9a-82da-3020328b9c2f.png)

3. Try to create association between both by soleProprietorship.
![image](https://user-images.githubusercontent.com/93864853/226987074-db339363-7f00-40f6-9f01-3df9a5a6948b.png)

**Expected behavior**
It's should be create the association between individual -> soleProprietorship. (Without showing any typescript error)

**Screenshots**
It's working perfect through normal api call. But the SDK missed the optional depends upon the type of LegalEntity.

![image (2)](https://user-images.githubusercontent.com/93864853/226987612-e9977a55-fea1-4e9a-872e-0d2b76c450e3.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Node Version: v18.12.1
 - NPM Version: v12.2.0

**Additional context**
We solved by force type. But it's successfully created the association. It would be good if no error to shown. 

We did something like this to make it work.

![image (1)](https://user-images.githubusercontent.com/93864853/226988073-5c8919aa-efed-4b97-8fdd-ddcf4c7a9a77.png)

Hope it'll be clear to fix the issue.

**Comment by ossiggy** [comment]
I believe the correct type here would be `LegalEntityAssociation.TypeEnum`

**Comment by MD-Fleksa** [comment]
Yeah that'll also work. If possible the documentation include that It'll be more helpful. Thank you the response.
